lib/resources/http_interceptors/auth_interceptor.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:http_interceptor/http_interceptor.dart';
+import 'package:wastexchange_mobile/resources/auth_manager.dart';
+
+/// Interceptor that modify API Request by adding authentication information to the request.
+class AuthInterceptor implements InterceptorContract {
+
+  @override
+  Future<RequestData> interceptRequest({RequestData data}) async {
+    await addAuthHeader(data);
+    return data;
+  }
+
+  @override
+  Future<ResponseData> interceptResponse({ResponseData data}) async {
+    return data;
+  }
+
+  Future<void> addAuthHeader(RequestData data) async {
+    final String token = await AuthManager().getAccessToken();
+    if(token != null) {
```

Comment:
It would be nice to have a mechanism where token == null throws an exception if an authenticated endpoint comes here and finds it to be null. Can we think of something?

Reply:
Throwing and exception is nice to have. The endpoint itself will throw 401 if token is missing.
An ideal state would be to prompt user to re-login and cache the new token.

Reply:
This we need to come up with a design. We may have endpoint that can work for both authenticated and non authenticated calls.

Reply:
Shall we go with https://pub.dev/packages/dio for HTTP calls. 
As they support more features like Retrying the request based on response, Canceling the request etc.

Reply:
Added two instance of ApiBaseHelper. One for unauthenticated calls and another for Authenticated calls.

Implementation can check the TokenRepository.isAuthorized() to decide to make unauthenticated call or not.

On 401/403 error, The logic is changed to perform token refresh, Retry internally and give the latest result.